nntrainer/layers/layer_node.h

```diff
@@ -103,49 +98,154 @@ class LayerNode final : public ml::train::Layer, public GraphNode {
+   * function must set output dimensions in the given context. Further, context
+   * can be used to request weights for the layer, and any extra tensor required
+   * for the operation of the layer.
+   * @note      After calling this it is not allowed to
+   * change properties.
+   * @note      No memory allocation must be performed in the initialization
+   * step. Any tensor memory required must be requested to the context which
+   * will be made available during execution of the layer with the context.
+   */
+  void finalize();
+
+  /**
+   * @brief     Forward Propagation of a layer
+   * @param     training true if training, false if inference
+   *
+   * @details   context provides access to the weights (if any), inputs,
+   * outputs, and tensors (if any) for the layer. Input and output dimensions
+   * can be access from the inputs/outputs tensors themselves.
+   */
+  void forwarding(bool training = true);
+
+  /**
+   * @brief     calc the derivative to be passed to the previous layer
+   *
+   * @details   context provides access to the weights (if any), inputs,
+   * outputs, and tensors (if any) for the layer. Input and output dimensions
+   * can be access from the inputs/outputs tensors themselves.
+   */
+  void calcDerivative();
+
+  /**
+   * @brief     Calculate the derivative of a layer
+   * @details   context provides access to the weights (if any), inputs,
+   * outputs, and tensors (if any) for the layer. Input and output dimensions
+   * can be access from the inputs/outputs tensors themselves.
+   */
+  void calcGradient();
+
+  /**
+   * @brief this function helps exporting the layer in a predefined format,
+   * while workarounding issue caused by templated function type eraser
+   *
+   * @param     exporter exporter that conatins exporting logic
+   * @param     method enum value to identify how it should be exported to
+   */
+  void exportTo(Exporter &exporter, const ExportMethods &method) const;
+
+  /**
+   * @brief Set the batch for the layer
+   * @param     context Context of the layer
+   * @param     batch Batch value to be set
+   * @details Update the run context based on the updated batch size if required
+   */
+  void setBatch(unsigned int batch);
+
+  /**
+   * @brief   If the current layer can support in-place
+   * @return  true if inplace, else false
+   */
+  bool supportInPlace() const;
+
+  /**
+   * @brief  check if this layer requires label to be passed
+   * @return true if requires a label when training, else false
+   * @note   if requireLabel() == true means, for now, that it is endpoint of a
+   * graph(numOutlayers == 0). label will be fed to the gradient of hidden if
+   * requireLabel is true
+   */
+  bool requireLabel() const;
+
+  /**
+   * Add rest of the helper interfaces required by other internal classes
    */
-  std::shared_ptr<nntrainer::LayerV1> &getObject();
 
   /**
    * @brief     Get underlying object
    *
+   * @todo      Deprecate this
```

Comment:
Suggesting to delete this, I have added as per comment so it will conflict

Reply:
will do.